### DISCUSSION TITLE
Faster builds and deployments in dev mode?

### DISCUSSION THREAD

**Discussion by aslom** [knative/func#1600] [Prototype Ideas]
I think that in "dev mode" we should allow as fast as possible incremental deployments and tests - code is compiled and streamed into target deployment during hot reload something like what Quarkus is doing?

https://quarkus.io/guides/dev-mode-differences
https://medium.com/geekculture/speed-up-development-and-save-time-with-live-reload-and-testing-in-quarkus-2-0-53af6ff65edb

That may be particularly interesting for Wasm as modules when compiled are small and could be streamed into target dpeoyment the same way that web browser do it today https://developer.mozilla.org/en-US/docs/WebAssembly/JavaScript_interface/instantiateStreaming

**Comment by ThaliaBumbray0mL** [comment]
Yes, using a dev mode that allows for fast incremental deployments and tests can be very beneficial for developers. The Quarkus framework's dev mode seems like a good example of this approach. Additionally, streaming compiled Wasm modules into the target deployment could also be an interesting approach, especially given the small size of Wasm modules. This could potentially improve the speed and efficiency of the development process.

**Comment by lkingland** [comment]
We are in the process of implementing the "Host" builder, which uses the current user's OS configuration for building, including the Function languages tooling (for example the `go` binary for Go, `node` for Node.js etc).  This will be able to be specified with the `--builder=host` flag.
 
This "host" build is performed without the use of a container, and is thus much faster.  This is intended for use in either development, or when the build environment can be easily configured.  For environments needing strict build isolation, are unconfigurable, or are in need of more advanced configuration, containerized builds using Buildpack (or S2I) will remain available.
